sdks/unity/AgonesSdk.cs

```diff
@@ -73,6 +73,31 @@ private void OnApplicationQuit()
         #endregion
 
         #region AgonesRestClient Public Methods
+
+        /// <summary>
+        /// Async method that waits to connect to the SDK Server. Will timeout
+        /// and return false after 30 seconds.
+        /// </summary>
+        /// <returns>A task that indicated whether it was successful or not</returns>
+        public async Task<bool> Connect()
+        {
+            for (var i = 0; i < 30; i++)
+            {
+                Log($"Attempting to connect...{i + 1}");
+                var gameServer = await GameServer();
+                if (gameServer != null)
+                {
+                    Log("Connected!");
+                    return true;
+                }
+                
+                Log("Connection failed, retrying.");

```

Comment:
Instead of delaying for a second, have you considered retrying in a loop with the condition that currentTime - startTime < 30 second?

Reply:
Interesting! Would that be a big CPU churn when the SDK hasn't spun up yet?

Reply:
You're right.